utest/utils/test_misc.py

```diff
@@ -42,6 +46,23 @@ def test_printable_name_with_code_style(self):
                          ('', '')]:
             assert_equals(printable_name(inp, code_style=True), exp)
 
+    def test_isatty(self):
+        assert_equals(isatty(sys.__stdout__), sys.__stdout__.isatty())
+        stream = io.StringIO()
+        assert_false(isatty(stream))
+        wrapper = io.TextIOWrapper(stream)
+        assert_false(isatty(wrapper))
+        # make sure that isatty() checks for detached buffers
+        # (otherwise it would raise a ValueError)
+        if sys.version_info >= (2, 7):
+            wrapper.detach() #==> wrapper.buffer is None
+            assert_raises(ValueError, lambda: wrapper.isatty())
+        else: # no .detach and different .isatty behaviour if .buffer is None
+            wrapper.buffer = None
+            assert_raises(AttributeError, lambda: wrapper.isatty())
+        assert_false(isatty(wrapper))
+        stream.close()
```

Comment:
+1 for a very thorough testing. It would be better to split this into multiple tests, one for each stream type. You can either keep them in the original test class and name them like `test_isatty_with_sys_stdout`, `test_isatty_with_stringio` or, preferably, create new `TestIsatty` class with individual tests like `test_sys_stdout`, `test_stringio`.